Symfony/CS/Fixer/Contrib/EregToPregFixer.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * This file is part of the Symfony CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class EregToPregFixer extends AbstractFixer
+{
+    /**
+     * @var array
+     */
+    private $map = array(
+        array('ereg', 'preg_match', ''),
+        array('eregi', 'preg_match', 'i'),
+        array('ereg_replace', 'preg_replace', ''),
+        array('eregi_replace', 'preg_replace', 'i'),
+        array('split', 'preg_split', ''),
+        array('spliti', 'preg_split', 'i'),
+    );
+
+    /**
+     * @var array
+     */
+    private $delimiters = array('/', '#', '!');
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        if (!$this->cursoryMatch($content)) {
+            return $content;
+        }
+
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($this->map as $map) {
+            $seq = array(array(T_STRING, $map[0]), '(');
+            $currIndex = 0;
+            while (null !== ($ereg = $tokens->findSequence($seq, $currIndex, $tokens->count(), false))) {
+                // Ensure it's a function call
```

Comment:
This logic could be extracted to the Tokens class (TokensAnalyzer in 2.0).
`isProceduralFunctionCall()`, `isObjectFunctionCall()`, `isStaticFunctionCall()`, `isInvokeFunctionCall()` so it can be reused.